core/src/mindustry/ui/fragments/ChatFragment.java

```diff
@@ -161,14 +162,15 @@ public void draw(){
 
         float theight = offsety + spacing + getMarginBottom() + scene.marginBottom;
         for(int i = scrollPos; i < messages.size && i < messagesShown + scrollPos && (i < fadetime || shown); i++){
+            String message = UI.formatIcons(messages.get(i));
```

Comment:
Don't change the way the chat displays. On desktop, you can already type :alpha: and have it show up as you type. Inject the formatting code into the chat message sending code instead, so it is formatted right before you send it.